homeassistant/components/hue_ble/light.py

```diff
@@ -0,0 +1,185 @@
+
+import logging
+from typing import Any
+
+from HueBLE import HueBleLight
+
+from homeassistant.components.light import (
+    ATTR_BRIGHTNESS,
+    ATTR_COLOR_TEMP,
+    ATTR_XY_COLOR,
+    ColorMode,
+    LightEntity,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.device_registry import CONNECTION_BLUETOOTH, DeviceInfo
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Add light for passed config_entry in HA."""
+
+    light = hass.data[DOMAIN][config_entry.entry_id]
+    entity = HaHueBLE(light)
+    async_add_entities([entity])
+
+
+class HaHueBLE(LightEntity):
+    """Representation of a light."""
+
+    def __init__(self, api: HueBleLight) -> None:
+        """Initialize the light object. Does not connect."""
+
+        self._light = api
+        self._name = self._light.name
+        self._address = self._light.address
+        self._attr_unique_id = self._light.address
+        self._attr_available = self._light.available
+        self._attr_name = self._name
+        self._attr_is_on = self._light.power_state
+        self._attr_brightness = self._light.brightness
+        self._attr_color_temp = self._light.colour_temp
+        self._attr_xy_color = self._light.colour_xy
+        self._attr_min_mireds = self._light.minimum_mireds
+        self._attr_max_mireds = self._light.maximum_mireds
+        self._attr_device_info = DeviceInfo(
+            connections={(CONNECTION_BLUETOOTH, self._light.address)},
+            identifiers={(DOMAIN, self._light.address)},
+            name=self._name,
+            manufacturer=self._light.manufacturer,
+            model=self._light.model,
+            sw_version=self._light.firmware,
+        )
+
+    async def async_added_to_hass(self) -> None:
+        """Run when this Entity has been added to HA."""
+
+        self._light.add_callback_on_state_changed(self._state_change_callback)
+
+    async def async_will_remove_from_hass(self) -> None:
+        """Run when entity will be removed from HA."""
+
+        self._light.remove_callback(self._state_change_callback)
+
+    def _state_change_callback(self) -> None:
+        """Run when light informs of state update. Updates local properties."""
+
+        _LOGGER.debug("Received state notification from light %s", self.name)
+
+        self._name = self._light.name
+        self._address = self._light.address
+        self._attr_available = self._light.available
+        self._attr_name = self._name
+        self._attr_is_on = self._light.power_state
+        self._attr_brightness = self._light.brightness
+        self._attr_color_temp = self._light.colour_temp
+        self._attr_xy_color = self._light.colour_xy
+
```

Comment:
You should use properties for all these value updates instead:

```
@property
def is_on(self) -> bool:
    """Return true if device is on."""
    return self._light.power_state
```

Also, don't update the name